app/api/actions/reservations.js

```diff
@@ -11,6 +14,21 @@ function fetchReservations(params = {}) {
   });
 }
 
+function makeReservation(reservation) {
+  return createApiAction({
+    endpoint: 'reservation',
+    method: 'POST',
+    type: 'RESERVATION',
+    body: parseReservationData(reservation),
+  });
+}
+
+function parseReservationData(reservation) {
+  const parsed = pickBy(reservation, value => value);
```

Comment:
I checked that the API doesn't work with null values either. But I changed the parse function to allow values that are 0.